microsoftgraph/microsoft-graph-devx-api
# Fix: Get all distinct permissions

Fixes https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/635
Fixes https://github.com/microsoftgraph/microsoft-graph-devx-api/issues/633

## Overview

Gets all distinct scopes from the permissions mapping files and from the Service Principal descriptions when servicing fetch-all permissions requests from both the [permissions-mapping](https://raw.githubusercontent.com/microsoftgraph/microsoft-graph-devx-content/dev/permissions/permissions-v1.0.json) and the [Service Principal](https://raw.githubusercontent.com/microsoftgraph/microsoft-graph-devx-content/dev/permissions/permissions-descriptions.json) descriptions.

### Notes

Resolves issues raised in this PR as well: https://github.com/microsoftgraph/microsoft-graph-devx-api/pull/628
@baywet I felt this syntax provides better readability and contextualization:
https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/9835dbb0e5b9410e3db8b9178124d443bd4fbe7b/GraphExplorerPermissionsService/Services/PermissionsStore.cs#L407-L411

## Testing Instructions

- Run: https://localhost:44399/permissions?scopeType=Application
- Search for: `Channel.Create.Group`
- You should be able to see this sample query from the list.
- Run: https://localhost:44399/permissions?requesturl=/me&method=GET
- Search for: `Directory.AccessAsUser.All`
- There should only be one instance of this.
